### ISSUE TITLE
Descriptor with `@staticmethod` `__set__` not modeled correctly

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4485]
I'm not sure why anyone would do that, but this works at runtime, and type-checks correctly in all other type checkers, whereas ty emits an error:

```py
class Descriptor:
    @staticmethod
    def __set__(instance: object, value: int) -> None:
        pass

class Owner:
    value = Descriptor()

owner = Owner()
owner.value = 1
```

```
error[invalid-assignment]: Invalid assignment to data descriptor attribute `value` on type `Owner`
  --> main.py:10:1
   |
10 | owner.value = 1
   | ^^^^^ Expected `int`, found `Owner`
info: Argument to function `Descriptor.__set__` is incorrect
info: This assignment implicitly calls `__set__` on a descriptor of type `Descriptor`
info: Function defined here
 --> main.py:3:9
  |
3 |     def __set__(instance: object, value: int) -> None:
  |         ^^^^^^^                   ---------- Parameter declared here

error[invalid-assignment]: Invalid assignment to data descriptor attribute `value` on type `Owner`
  --> main.py:10:15
   |
10 | owner.value = 1
   |               ^ Too many positional arguments to function `Descriptor.__set__`: expected 2, got 3
info: This assignment implicitly calls `__set__` on a descriptor of type `Descriptor`
info: Function signature here
 --> main.py:3:9
  |
3 |     def __set__(instance: object, value: int) -> None:
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

````
